app/Jobs/ImportCSV.php

```diff
@@ -46,16 +46,63 @@ public function handle(CSVImportReader $reader)
         $filepath = Storage::disk('local')
             ->path('mismatch-files/' . $this->meta->filename);
 
-        DB::transaction(function () use ($reader, $filepath) {
-            $reader->lines($filepath)->each(function ($mismatchLine) {
-                $mismatch = Mismatch::make($mismatchLine);
-                if ($mismatch->type == null) {
-                    $mismatch->type = 'statement';
+        $mismatch_attrs = (new Mismatch())->getFillable();
+
+        DB::transaction(function () use ($reader, $filepath, $mismatch_attrs) {
+            $new_mismatches = [];
+            $where_clauses = [];
+
+            $mismatches_per_upload_user = DB::table('mismatches')
+                ->select($mismatch_attrs)
+                ->join('import_meta', 'mismatches.import_id', '=', 'import_meta.id')
+                ->where('import_meta.user_id', '=', $this->meta->user->id);
+
+            $reader->lines($filepath)->each(function ($mismatchLine) use (
+                $mismatches_per_upload_user,
+                &$new_mismatches,
+                &$where_clauses
+            ) {
+
+                $new_mismatch = Mismatch::make($mismatchLine);
+                $new_mismatches[] = $new_mismatch;
+                $collection = collect($new_mismatch->getAttributes());
+                $collection->forget('review_status');
+                $newArray = [['review_status', '!=', 'pending']];
+                $collection->map(function ($item, $key) use (&$newArray) {
+                    if ($key != 'type') { // key can be empty in the file but in the db always has statement by default
+                        $newArray[] = [$key, $item];
```

Comment:
Are we sure we need `$newArray`? instead of assigning directly to the `$where_clauses` in the loop, that is. Would it effect performance somehow?